nomic/atlas.py

```diff
@@ -2,1315 +2,234 @@
-                                         projection_spread=projection_spread,
-                                         multilingual=multilingual)
-            return dict(response)
-        else:
-            # otherwise refresh the maps
-            self.refresh_maps(project_id=project_id)
-            return {'project_name': project_name,
-                    'project_id': project_id}
-
-        return dict(response)
-
-
-
-
-    def delete_project(self, project_id: str):
-        '''
-        Deletes an atlas project with all associated metadata.
-
-        **Parameters:**
-
-        * **project_id** - The id of the project you want to delete.
-        '''
-        organization = self._get_current_users_main_organization()
-        organization_name = organization['nickname']
-
-        project = self._get_project_by_id(project_id=project_id)
-
-
-        logger.info(f"Deleting project `{project['project_name']}` from organization `{organization_name}`")
-
-        response = requests.post(
-            self.atlas_api_path + "/v1/project/remove",
-            headers=self.header,
-            json={'project_id': project_id},
-        )
-
-    def download_embeddings(self, project_id, atlas_index_id, output_dir, num_workers=10):
-        '''
-        Downloads a mapping from datum_id to embedding in shards to the provided directory
-
-        Args:
-            project_id: the id of the relevant index's parent project
-            atlas_index_id: the id of the index whose ambient embeddings you want
-            output_dir: the directory to save shards to
-        '''
-
-        response = requests.get(
-            self.atlas_api_path + f"/v1/project/{project_id}",
-            headers=self.header,
-        )
-        project = response.json()
-
-        total_datums = project['total_datums_in_project']
-        if project['insert_update_delete_lock']:
-            raise Exception('Project is locked! Please wait until the project is unlocked to download embeddings')
+            except BaseException as e:
+                if number_of_datums_before_upload == 0:
+                    logger.info("Deleting project due to failure in initial upload.")
+                    project.delete()
+                raise e
+
+    logger.info("Embedding upload succeeded.")
+
+    # make a new index if there were no datums in the project before
+    if number_of_datums_before_upload == 0:
+        projection = project.create_index(
+            name=index_name,
+            colorable_fields=colorable_fields,
+            build_topic_model=build_topic_model,
+            projection_n_neighbors=projection_n_neighbors,
+            projection_epochs=projection_epochs,
+            projection_spread=projection_spread,
+            topic_label_field=topic_label_field,
+        )
+        logger.info(str(projection))
+    else:
+        # otherwise refresh the maps
+        project.refresh_maps()
```

Comment:
not sold on this auto refresh but i dont have a strong case against so prob fine for now